### ISSUE TITLE
error when converting onnx to openvion

### ISSUE THREAD

**Issue by dandanli814** [openvinotoolkit/training_extensions#281]
Thank you for your work.
When after finetuning the model using NNCF, I want to transfer the .pth model to onnx and then convert onnx to openvion, an error occurred, as follows.

First of all, when I was finetuning, the h and w are scaled to a fixed scale like [b, c, 64, 1024]. But when I want to inference, the input shape is [b, c, 64, -1],  -1 means that the width will maintain the same scaling factor with height.
So based on that, first I modified your code https://github.com/opencv/openvino_training_extensions/blob/develop/pytorch_toolkit/nncf/nncf/compression_method_api.py#L166 as following:

with torch.no_grad():
        torch.onnx.export(model, tuple(input_tensor_list),
                                    filename, verbose=True, input_names = ['input'],
                                    dynamic_axes={'input':[3]})

Then I use your code to transfer my .pth model to onnx. Then I run this:

python3 /opt/intel/openvino/deployment_tools/model_optimizer/mo.py --input_model ./results/my_int8.onnx --output_dir ./results/

But an error occurs as follows:

Model Optimizer arguments:
Common parameters:
        - Path to the Input Model:      ./results/my_int8.onnx
        - Path for generated IR:        ./results/
        - IR output name:       my_int8
        - Log level:    ERROR
        - Batch:        Not specified, inherited from the model
        - Input layers:         Not specified, inherited from the model
        - Output layers:        Not specified, inherited from the model
        - Input shapes:         Not specified, inherited from the model
        - Mean values:  Not specified
        - Scale values:         Not specified
        - Scale factor:         Not specified
        - Precision of IR:      FP32
        - Enable fusing:        True
        - Enable grouped convolutions fusing:   True
        - Move mean values to preprocess section:       False
        - Reverse input channels:       False
ONNX specific parameters:
Model Optimizer version:        2020.1.0-61-gd349c3ba4a
[ ERROR ]  Shape [  1 128   4  -1] is not fully defined for output 0 of "1385". Use --input_shape with positive integers to override model input shapes.
[ ERROR ]  Cannot infer shapes or values for node "1385".
[ ERROR ]  Not all output shapes were inferred or fully defined for node "1385".
 For more information please refer to Model Optimizer FAQ (https://docs.openvinotoolkit.org/latest/_docs_MO_DG_prepare_model_Model_Optimizer_FAQ.html), question #40.
[ ERROR ]
[ ERROR ]  It can happen due to bug in custom shape infer function <function Elementwise.__init__.<locals>.<lambda> at 0x7f3b17b0c488>.
[ ERROR ]  Or because the node inputs have incorrect values/shapes.
[ ERROR ]  Or because input shapes are incorrect (embedded to the model or passed via --input_shape).
[ ERROR ]  Run Model Optimizer with --log_level=DEBUG for more information.
[ ERROR ]  Exception occurred during running replacer "REPLACEMENT_ID" (<class 'extensions.middle.PartialInfer.PartialInfer'>): Stopped shape/value propagation at "1385" node.
 For more information please refer to Model Optimizer FAQ (https://docs.openvinotoolkit.org/latest/_docs_MO_DG_prepare_model_Model_Optimizer_FAQ.html), question #38.

I don't know where the problem is. I hope your can help me. Thank you.

**Comment by AlexanderDokuchaev** [comment]
@AlexKoff88 @vshampor @ljaljushkin

**Comment by vshampor** [comment]
Greetings, @dandanli814 !

Sadly, AFAIK Model Optimizer does not support dynamic model inputs when generating IR. If you really want your OpenVINO-powered model to infer inputs with different shapes in a target application, you should instead use Inference Engine's model reshaping mechanisms each time your input shape is changed. See https://docs.openvinotoolkit.org/latest/_docs_IE_DG_ShapeInference.html and general OpenVINO documentation for more details.

I would further suggest that you first try to enable your model execution with OpenVINO using static input shapes in ONNX (if dynamic inputs are not critical for your model or scenario), and then, once the static input model is working, experiment with reshaping.

**Comment by dandanli814** [comment]
OK, I will try the static input model first, thank you very much.

**Comment by dandanli814** [comment]
@vshampor Hi, thank you for your work. Now, I want to inference your mobilenetv2_imagenet_int8.pth (https://download.01.org/opencv/openvino_training_extensions/models/nncf/mobilenetv2_imagenet_int8.pth) for classification with OPENVINO. I run the following commands:

1、python main.py -m test --config ./configs/quantization/mobilenetv2_imagenet_int8.json --resume mobilenetv2_imagenet_int8.pth --to-onnx mobilenetV2_int8.onnx

2、python3 /opt/intel/openvino/deployment_tools/model_optimizer/mo.py --input_model mobilenetV2_int8.onnx --output_dir ./ --mean_values [0.485,0.456,0.406] --scale_values [0.229,0.224,0.225]

3、python classification_sample.py -m mobilenetV2_int8.xml -i car.jpg --labels imagenet1000_clsidx_to_labels.txt

where the classification_sample.py is provided by OPENVINO in “/opt/intel/openvino_2020.1.023/deployment_tools/inference_engine/samples/python/classification_sample/classification_sample.py”, the --labels file is from "https://gist.github.com/yrevar/667fd94b94f1666137f45d1363f60910", the -i car.jpg is the following:
![car](https://user-images.githubusercontent.com/24379250/80195414-60edf500-864e-11ea-9270-cadf57b3559a.jpg)

Then the classification result is as following:

[ INFO ] Creating Inference Engine
[ INFO ] Loading network files:
        results/mobilenetV2_int8.xml
        results/mobilenetV2_int8.bin
[ INFO ] Preparing input blobs
[ WARNING ] Image car.jpg is resized from (720, 1280) to (224, 224)
[ INFO ] Batch size is 1
[ INFO ] Loading model to the plugin
[ INFO ] Starting inference in synchronous mode
[ INFO ] Processing output blob
[ INFO ] Top 10 results:
Image car.jpg
classid probability
------- -----------
549: 'envelope', 10.5345688
769: 'rule, ruler', 10.4771109
904: 'window screen', 10.0740719
611: 'jigsaw puzzle', 9.6414747
709: 'pencil box, pencil case', 9.1171026
893: 'wallet, billfold, notecase, pocketbook', 7.4382539
443: 'bib', 7.3135014
584: 'hair slide', 7.2139525
721: 'pillow', 6.8544011
600: 'hook, claw', 6.7706876

It is wrong. I don't know where the problem is. Looking forward to your advice.
